tests/api/compiler.cpp

```diff
@@ -35,3 +35,61 @@ TEST_F(WithContext, BuildLog) {
     build_log = GetProgramBuildLog(program_error);
     ASSERT_TRUE(build_log.find("THIS IS AN ERROR") != std::string::npos);
 }
+
+TEST_F(WithCommandQueue, SimpleLink) {
```

Comment:
Could you add a short comment describing what is being tested please? Maybe a test name change is good enough. `CompileAndLinkWithPushConstants`?